api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -153,4 +153,9 @@ message TcpProxy {
   // is reached the connection will be closed. Duration must be at least 1ms.
   google.protobuf.Duration max_downstream_connection_duration = 13
       [(validate.rules).duration = {gte {nanos: 1000000}}];
+
+  // The maximum duration tcp_proxy waits for the active cluster. The duration is counted since the tcp_proxy starts to process the connection.
```

Comment:
I'm hesitant to add this field now without a better understanding of what we are actually going to end up implementing. Let's wait until there is a more fully formed design doc that is agreed upon or there is an actual PR to look at. Thanks!

/wait